accounts-db/src/tiered_storage/readable.rs

```diff
@@ -72,12 +75,23 @@ impl<'accounts_file, M: TieredAccountMeta> ReadableAccount
     }
 
     /// Returns the epoch that this account will next owe rent by parsing
-    /// the specified account block.  Epoch::MAX will be returned if the account
-    /// is rent-exempt.
+    /// the specified account block.  RENT_EXEMPT_RENT_EPOCH will be returned
+    /// if the account is rent-exempt.
+    ///
+    /// For a zero-lamport account, Epoch::default() will be returned to
+    /// default states of an AccountSharedData.
     fn rent_epoch(&self) -> Epoch {
         self.meta
             .rent_epoch(self.account_block)
-            .unwrap_or(Epoch::MAX)
+            .unwrap_or(if self.lamports() != 0 {
+                RENT_EXEMPT_RENT_EPOCH
+            } else {
+                // While there is no valid-values for any fields of a zero
+                // lamport account, here we return Epoch::default() to
+                // match the default states of AccountSharedData.  Otherwise,
+                // a hash mismatch will occur.
+                Epoch::default()
+            })
     }
```

Comment:
@yhchiang-sol are we going to also change what gets written:
https://github.com/jeffwashington/solana/blob/ef6c506ab673c570220f39ba2830cd3ce942a40d/accounts-db/src/tiered_storage/hot.rs#L593

Reply:
writing a 0 lamport account would currently store the rent_epoch as a 0 u64

Reply:
> writing a 0 lamport account would currently store the rent_epoch as a 0 u64

It shouldn't be the case because zero lamport account will be None:
https://github.com/jeffwashington/solana/blob/ef6c506ab673c570220f39ba2830cd3ce942a40d/accounts-db/src/account_storage/meta.rs#L81


And, for None case, we will write None for the rent_epoch
https://github.com/jeffwashington/solana/blob/ef6c506ab673c570220f39ba2830cd3ce942a40d/accounts-db/src/tiered_storage/hot.rs#L597